cmake/cuda.cmake

```diff
@@ -0,0 +1,78 @@
+enable_language(CUDA)
+
+if(MSVC)
+    # MSVC can not find CUDA automatically
+    # Use CUDA_COMPILER PATH to define the CUDA TOOLKIT ROOT DIR
+    string(REPLACE "/bin/nvcc.exe" "" CMAKE_CUDA_ROOT_DIR ${CMAKE_CUDA_COMPILER})
+    if("${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES}" STREQUAL "")
+        set(CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES "${CMAKE_CUDA_ROOT_DIR}/include")
+    endif()
+    if("${CMAKE_CUDA_IMPLICIT_LINK_DIRECTORIES}" STREQUAL "")
+        set(CMAKE_CUDA_IMPLICIT_LINK_DIRECTORIES "${CMAKE_CUDA_ROOT_DIR}/lib/x64")
+    endif()
+endif()
+
+include(cmake/Modules/CudaArchitectureSelector.cmake)
```

Comment:
since this is now being called from the root CMakeLists.txt instead of the cuda CMakeLists.txt, I would prefer removing the third_party link to CAS, yes. Do you see any benefit from keeping it (also allowing external CAS, as discussed elsewhere)?

Reply:
As our minimum CMake version grows, we will probably move to CMake's CUDA_ARCHITECTURES target property at some point, anyways? I think CAS will be slimmed down significantly then.